comp/core/autodiscovery/autoconfig.go

```diff
@@ -55,6 +73,31 @@ type AutoConfig struct {
 	ranOnce *atomic.Bool
 }
 
+type provides struct {
+	fx.Out
+
+	Comp         Component
+	OptionalComp optional.Option[Component]
```

Comment:
Why are we returning a component instance and optional component instance?

Reply:
Because  the dependent of autodiscovery like flare component needs optional component instance, on the other hand, agent needs a non-optional AD component. This implementation is very similar to collector component.